### PR TITLE
fix(delibird): pass stdin to subproccess when not term

### PR DESCRIPTION
**Pull request by jaredallard**
## What this PR does / why we need it

This PR fixes the stdin logic to properly pass stdin to the created sub-process when running not as a terminal. When running in a terminal, this is properly handled by the `ptyOutputHook` function.

## Jira ID

[DT-3742]

[DT-3742]: https://outreach-io.atlassian.net/browse/DT-3742?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: pkg/cli/logfile/logfile_unix.go [hunk 1]
```diff
@@ -101,6 +101,7 @@ func Hook() error {
 
 		cmd.Stdout = io.MultiWriter(os.Stdout, rec)
 		cmd.Stderr = io.MultiWriter(os.Stderr, rec)
+		cmd.Stdin = os.Stdin
```
Comment (jkinkead) [inline pkg/cli/logfile/logfile_unix.go:104]: The error was being triggered on line 84, so this isn't fixing the bug you think it is.

**Review by jkinkead — APPROVED** [review]
Changes look good, but this isn't fixing the root cause of the terminal width not being discoverable.

File: pkg/cli/logfile/logfile_unix.go [hunk 2]
```diff
@@ -33,7 +33,8 @@ func Hook() error {
 		return nil
 	}
 
-	isTerminal := term.IsTerminal(int(os.Stdout.Fd()))
+	// if both stdin and stdout are not terminals, then we don't need a pty
```
Comment (jkinkead) [inline pkg/cli/logfile/logfile_unix.go:36]: Why do we need a terminal _at all_? I can't remember this.

**Review by jkinkead — COMMENTED** [review]
Is there any way to plug in a simple end-to-end test here? Not an Outreach "e2e", but like:

1. Build a simple binary that uses this library.
2. Verify that you can pipe stuff to it / redirect from it.

File: pkg/cli/logfile/logfile_unix.go (hunk 2 above)
Comment (jaredallard) [reply pkg/cli/logfile/logfile_unix.go:36]: If we pipe to a file, terminal checks will fail. This means that logs will never be formatted. We use a pty to ensure that users can still use things that need user input and terminal drawing (e.g., `devenv top`) and have nicely formatted logs.

**Comment by jaredallard** [comment]
> Is there any way to plug in a simple end-to-end test here? Not an Outreach "e2e", but like:
> 
> 1. Build a simple binary that uses this library.
> 2. Verify that you can pipe stuff to it / redirect from it.

@jkinkead  Yes, we could add it. I'd prefer to do it in another ticket though given the rest of this functionality has been tested already and this fixes a small bug in the original implementation. (I'll pick it up soon but don't want to block all of this on that, if possible.

**Review by jkinkead — APPROVED** [review]
LGTM, thanks for the answers!

**Review by tonymbugua-outreach — APPROVED** [review]

**Pull request merged**